src/main/java/com/accesspoint/rulesengine/service/RulesetService.java

```diff
@@ -69,10 +64,84 @@ public ResponseEntity<Ruleset> createRuleset(Ruleset incomingRulesetData) {
                         .rules(incomingRulesetData.getRules())
                 .build()
         );
-
         return ResponseEntity.ok().body(this.rulesetRepository.getReferenceById(rulesetData.getId()));
     }
 
+    public ResponseEntity<Ruleset> updateRuleset(Long id, Ruleset incomingRuleset) {
+        if (incomingRuleset.getName() == null || incomingRuleset.getName().isEmpty()) throw new BadRequestException("Name cannot be empty");
+        if (incomingRuleset.getRules() == null || incomingRuleset.getRules().isEmpty()) throw new BadRequestException("Rules cannot be empty");
+        incomingRuleset.getRules().stream().forEach(rule -> {
+            if (rule.getPriority() == 0.0) throw new BadRequestException("Rule priority cannot be 0");
+            if (rule.getEvent_type() == null) throw new BadRequestException("Rule event type cannot be null");
+            if (rule.getConditions() == null || rule.getConditions().isEmpty()) throw new BadRequestException("Conditions cannot be empty");
+            rule.getConditions().stream().forEach(condition -> {
+                if (condition.getFact_type() == null) throw new BadRequestException("Condition fact type cannot be null");
+                if (condition.getValue_type() == null) throw new BadRequestException("Condition value type cannot be null");
+            });
+        });
+
+
+        Ruleset updatedRuleset2 = rulesetRepository.findById(id) //
+                .map(oldRuleset -> {
+                    //TODO: optimize Hibernate select calls
```

Comment:
Comment 1:
Remove "TODO" if implemented or explain comment

Reply:
needs to be left in for future implementation